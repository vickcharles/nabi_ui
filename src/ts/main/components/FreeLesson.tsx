import * as React from 'react';
import { Button } from 'material-ui';

/** 
 * FreeLesson callout in homepage
 */
const FreeLesson = () => {
  return (
    <div className="nabi-background-nabi free-lesson-section nabi-color-white">
        <div className="nabi-container nabi-padding-top-large nabi-padding-bottom-large nabi-position-relative">
        <div className="free-lesson-left-section">
        <p className="nabi-margin-bottom-xsmall nabi-margin-top-zero nabi-font-large"> 
          Get stared with a free lesson today
        </p>
        <p className="nabi-margin-bottom-zero nabi-font-medium">
          Learn music with your city's most talented and qualified music instructors
        </p>
        <Button className="nabi-margin-top-xsmall" color="secondary" variant="raised">
          Try your FREE lesson
        </Button>
        </div>
        <div className="free-lesson-right-section nabi-margin-top-medium">
          <img 
            src={require('../../../assets/images/nabi-home-music-icon.png')} 
            alt=""
            className="free-lesson-image"
          />
          <p className="nabi-font-jennasue nabi-margin-top-zero free-lesson-lema">
            Share the love for music thought education
          </p>
        </div>
       </div>
    </div>
  );
};

export default FreeLesson;
