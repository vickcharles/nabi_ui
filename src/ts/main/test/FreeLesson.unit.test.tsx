import * as React from 'react';
import FreeLesson from '../components/FreeLesson';
import { shallow } from 'enzyme';

describe('FreeLesson', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(
      <FreeLesson />
    );
  });

  it('Renders the component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
