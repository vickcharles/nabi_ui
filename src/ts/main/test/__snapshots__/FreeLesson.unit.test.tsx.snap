// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FreeLesson Renders the component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FreeLesson />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="nabi-container nabi-padding-top-large nabi-padding-bottom-large nabi-position-relative"
      >
        <div
          className="free-lesson-left-section"
        >
          <p
            className="nabi-margin-bottom-xsmall nabi-margin-top-zero nabi-font-large"
          >
            Get stared with a free lesson today
          </p>
          <p
            className="nabi-margin-bottom-zero nabi-font-medium"
          >
            Learn music with your city's most talented and qualified music instructors
          </p>
          <Link
            replace={false}
            to="registration-student"
          >
            <WithStyles(Button)
              className="nabi-margin-top-xsmall"
              color="secondary"
              variant="raised"
            >
              Try your FREE lesson
            </WithStyles(Button)>
          </Link>
        </div>
        <div
          className="free-lesson-right-section nabi-margin-top-medium"
        >
          <img
            alt=""
            className="free-lesson-image"
            src="nabi-home-music-icon.png"
          />
          <p
            className="nabi-font-jennasue nabi-margin-top-zero free-lesson-lema"
          >
            Share the love for music thought education
          </p>
        </div>
      </div>,
      "className": "nabi-background-nabi free-lesson-section nabi-color-white",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="free-lesson-left-section"
          >
            <p
              className="nabi-margin-bottom-xsmall nabi-margin-top-zero nabi-font-large"
            >
              Get stared with a free lesson today
            </p>
            <p
              className="nabi-margin-bottom-zero nabi-font-medium"
            >
              Learn music with your city's most talented and qualified music instructors
            </p>
            <Link
              replace={false}
              to="registration-student"
            >
              <WithStyles(Button)
                className="nabi-margin-top-xsmall"
                color="secondary"
                variant="raised"
              >
                Try your FREE lesson
              </WithStyles(Button)>
            </Link>
          </div>,
          <div
            className="free-lesson-right-section nabi-margin-top-medium"
          >
            <img
              alt=""
              className="free-lesson-image"
              src="nabi-home-music-icon.png"
            />
            <p
              className="nabi-font-jennasue nabi-margin-top-zero free-lesson-lema"
            >
              Share the love for music thought education
            </p>
          </div>,
        ],
        "className": "nabi-container nabi-padding-top-large nabi-padding-bottom-large nabi-position-relative",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="nabi-margin-bottom-xsmall nabi-margin-top-zero nabi-font-large"
              >
                Get stared with a free lesson today
              </p>,
              <p
                className="nabi-margin-bottom-zero nabi-font-medium"
              >
                Learn music with your city's most talented and qualified music instructors
              </p>,
              <Link
                replace={false}
                to="registration-student"
              >
                <WithStyles(Button)
                  className="nabi-margin-top-xsmall"
                  color="secondary"
                  variant="raised"
                >
                  Try your FREE lesson
                </WithStyles(Button)>
              </Link>,
            ],
            "className": "free-lesson-left-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Get stared with a free lesson today",
                "className": "nabi-margin-bottom-xsmall nabi-margin-top-zero nabi-font-large",
              },
              "ref": null,
              "rendered": "Get stared with a free lesson today",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Learn music with your city's most talented and qualified music instructors",
                "className": "nabi-margin-bottom-zero nabi-font-medium",
              },
              "ref": null,
              "rendered": "Learn music with your city's most talented and qualified music instructors",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(Button)
                  className="nabi-margin-top-xsmall"
                  color="secondary"
                  variant="raised"
                >
                  Try your FREE lesson
                </WithStyles(Button)>,
                "replace": false,
                "to": "registration-student",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Try your FREE lesson",
                  "className": "nabi-margin-top-xsmall",
                  "color": "secondary",
                  "variant": "raised",
                },
                "ref": null,
                "rendered": "Try your FREE lesson",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt=""
                className="free-lesson-image"
                src="nabi-home-music-icon.png"
              />,
              <p
                className="nabi-font-jennasue nabi-margin-top-zero free-lesson-lema"
              >
                Share the love for music thought education
              </p>,
            ],
            "className": "free-lesson-right-section nabi-margin-top-medium",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "className": "free-lesson-image",
                "src": "nabi-home-music-icon.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Share the love for music thought education",
                "className": "nabi-font-jennasue nabi-margin-top-zero free-lesson-lema",
              },
              "ref": null,
              "rendered": "Share the love for music thought education",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="nabi-container nabi-padding-top-large nabi-padding-bottom-large nabi-position-relative"
        >
          <div
            className="free-lesson-left-section"
          >
            <p
              className="nabi-margin-bottom-xsmall nabi-margin-top-zero nabi-font-large"
            >
              Get stared with a free lesson today
            </p>
            <p
              className="nabi-margin-bottom-zero nabi-font-medium"
            >
              Learn music with your city's most talented and qualified music instructors
            </p>
            <Link
              replace={false}
              to="registration-student"
            >
              <WithStyles(Button)
                className="nabi-margin-top-xsmall"
                color="secondary"
                variant="raised"
              >
                Try your FREE lesson
              </WithStyles(Button)>
            </Link>
          </div>
          <div
            className="free-lesson-right-section nabi-margin-top-medium"
          >
            <img
              alt=""
              className="free-lesson-image"
              src="nabi-home-music-icon.png"
            />
            <p
              className="nabi-font-jennasue nabi-margin-top-zero free-lesson-lema"
            >
              Share the love for music thought education
            </p>
          </div>
        </div>,
        "className": "nabi-background-nabi free-lesson-section nabi-color-white",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="free-lesson-left-section"
            >
              <p
                className="nabi-margin-bottom-xsmall nabi-margin-top-zero nabi-font-large"
              >
                Get stared with a free lesson today
              </p>
              <p
                className="nabi-margin-bottom-zero nabi-font-medium"
              >
                Learn music with your city's most talented and qualified music instructors
              </p>
              <Link
                replace={false}
                to="registration-student"
              >
                <WithStyles(Button)
                  className="nabi-margin-top-xsmall"
                  color="secondary"
                  variant="raised"
                >
                  Try your FREE lesson
                </WithStyles(Button)>
              </Link>
            </div>,
            <div
              className="free-lesson-right-section nabi-margin-top-medium"
            >
              <img
                alt=""
                className="free-lesson-image"
                src="nabi-home-music-icon.png"
              />
              <p
                className="nabi-font-jennasue nabi-margin-top-zero free-lesson-lema"
              >
                Share the love for music thought education
              </p>
            </div>,
          ],
          "className": "nabi-container nabi-padding-top-large nabi-padding-bottom-large nabi-position-relative",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="nabi-margin-bottom-xsmall nabi-margin-top-zero nabi-font-large"
                >
                  Get stared with a free lesson today
                </p>,
                <p
                  className="nabi-margin-bottom-zero nabi-font-medium"
                >
                  Learn music with your city's most talented and qualified music instructors
                </p>,
                <Link
                  replace={false}
                  to="registration-student"
                >
                  <WithStyles(Button)
                    className="nabi-margin-top-xsmall"
                    color="secondary"
                    variant="raised"
                  >
                    Try your FREE lesson
                  </WithStyles(Button)>
                </Link>,
              ],
              "className": "free-lesson-left-section",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Get stared with a free lesson today",
                  "className": "nabi-margin-bottom-xsmall nabi-margin-top-zero nabi-font-large",
                },
                "ref": null,
                "rendered": "Get stared with a free lesson today",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Learn music with your city's most talented and qualified music instructors",
                  "className": "nabi-margin-bottom-zero nabi-font-medium",
                },
                "ref": null,
                "rendered": "Learn music with your city's most talented and qualified music instructors",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(Button)
                    className="nabi-margin-top-xsmall"
                    color="secondary"
                    variant="raised"
                  >
                    Try your FREE lesson
                  </WithStyles(Button)>,
                  "replace": false,
                  "to": "registration-student",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Try your FREE lesson",
                    "className": "nabi-margin-top-xsmall",
                    "color": "secondary",
                    "variant": "raised",
                  },
                  "ref": null,
                  "rendered": "Try your FREE lesson",
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt=""
                  className="free-lesson-image"
                  src="nabi-home-music-icon.png"
                />,
                <p
                  className="nabi-font-jennasue nabi-margin-top-zero free-lesson-lema"
                >
                  Share the love for music thought education
                </p>,
              ],
              "className": "free-lesson-right-section nabi-margin-top-medium",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "free-lesson-image",
                  "src": "nabi-home-music-icon.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Share the love for music thought education",
                  "className": "nabi-font-jennasue nabi-margin-top-zero free-lesson-lema",
                },
                "ref": null,
                "rendered": "Share the love for music thought education",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
